internal/pkg/argocd/argocd_test.go

```diff
@@ -273,3 +275,39 @@ func TestFindArgocdAppByPathAnnotationRelative2(t *testing.T) {
 		t.Errorf("App name is not right-app")
 	}
 }
+
+func TestFindArgocdAppByPathAnnotationNotFound(t *testing.T) {
+	t.Parallel()
+	defer testutils.Quiet()()
+	ctx := context.Background()
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	mockApplicationClient := mocks.NewMockApplicationServiceClient(ctrl)
+	expectedResponse := &argoappv1.ApplicationList{
+		Items: []argoappv1.Application{
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						"argocd.argoproj.io/manifest-generate-paths": "wrong-path",
+					},
+					Name: "right-app",
+				},
+				Spec: argoappv1.ApplicationSpec{
+					Source: &argoappv1.ApplicationSource{
+						RepoURL: "",
+						Path:    "right/",
+					},
+				},
+			},
+		},
+	}
+
+	mockApplicationClient.EXPECT().List(gomock.Any(), gomock.Any()).Return(expectedResponse, nil)
+	app, err := findArgocdAppByManifestPathAnnotation(ctx, "right/path", "some-repo", mockApplicationClient)
```

Comment:
Is the reason this doesn't return an app because `right/` !=  `right/path`?
If so I would use a different string to make it extra clear

Reply:
like `Non existing app path` or something like that